### ISSUE TITLE
Implement aggregation operators

### ISSUE THREAD

**Issue by szarnyasg** [FTSRG/ingraph#82]
Aggregration operators and statistical functions behave similarly.

The list of [aggregation functions](https://neo4j.com/docs/developer-manual/3.0/cypher/#query-clause) in Cypher is the following:
* [x] [`avg()`](https://neo4j.com/docs/developer-manual/3.0/cypher/#aggregation-avg)
* [x] [`count()`](https://neo4j.com/docs/developer-manual/3.0/cypher/#_count)
* [x] [`max()`](https://neo4j.com/docs/developer-manual/3.0/cypher/#aggregation-max)
* [x] [`min()`](https://neo4j.com/docs/developer-manual/3.0/cypher/#aggregation-min)
* [x] [`sum()`](https://neo4j.com/docs/developer-manual/3.0/cypher/#aggregation-sum)
* [x] [`collect()`](https://neo4j.com/docs/developer-manual/3.0/cypher/#aggregation-collect)

For statistical computations, we can use the Apache Commons Math library: http://commons.apache.org/proper/commons-math/javadocs/api-3.6/org/apache/commons/math3/stat/descriptive/moment/StandardDeviation.html

* [ ] ~[`stdDev()`](https://neo4j.com/docs/developer-manual/3.0/cypher/#aggregation-stdev)~
* [ ] ~[`stdDevP()`](https://neo4j.com/docs/developer-manual/3.0/cypher/#aggregation-stdevp)~
* [ ] ~[`percentileCont()`](https://neo4j.com/docs/developer-manual/3.0/cypher/#aggregation-percentilecont)~
* [ ] ~[`percentileDisc()`](https://neo4j.com/docs/developer-manual/3.0/cypher/#aggregation-percentiledisc)~

It's important to note that all of these functions are calculated on _a property_, i.e. they do not take a list as an input. So these should not be implemented as functions, instead they should be implemented as Rete nodes.

Some snippets that we can use:

```scala
case PERCENTILECONT => (x) => percentile.evaluate()
case PERCENTILEDISC => (x) => percentile.evaluate()

case STDDEV => (list) => stddev.evaluate(list.asInstanceOf[Seq[Double]].toArray)
case STDDEVP => (list) => stddevp.evaluate(list.asInstanceOf[Seq[Double]].toArray)

val stddev = new StandardDeviation(false)
val stddevp = new StandardDeviation(true)
val percentile = new Percentile()
```

**Comment by szarnyasg** [comment]
Example data:

| a | b | c |
| --- | --- | --- |
| x | 1 | 91 |
| x | 2 | 92 |
| y | 3 | 93 |

projection_{a, count(b) -> count, min(c) -> min}

| a | count | min |
| --- | --- | --- |
| x | 2 | 91 |
| y | 3 | 93 |

**Comment by szarnyasg** [comment]
We discussed with @jmarton yesterday that we for the physical query plan, we should merge subsequent grouping and projection operators to a single operator (similarly to the `SkipSortOperator`).

**Comment by szarnyasg** [comment]
The transformation for mergint Grouping and Projection operators is implemented in 26d5cf4b756c20eb3e33ccc5a2db815b30b9b247
